Essentials/src/com/earth2me/essentials/ReserveEssentials.java

```diff
@@ -0,0 +1,1830 @@
+     */
+    @Override
+    public String version() {
+        return "0.1.2.0";
+    }
+
+    //This is our method to convert UUID -> username for use with Essentials' create account methods.
+    private String getName(UUID identifier) {
+        final User user = plugin.getUser(identifier);
+        return ((user == null)? identifier.toString() : user.getName());
+    }
+
+    /**
+     * @return Whether or not this implementation is enabled.
+     */
+    @Override
+    public boolean enabled() {
+        return true;
+    }
+
+    /**
+     * @return Whether or not this implementation should have a default Vault implementation.
+     */
+    @Override
+    public boolean vault() {
+        return false;
+    }
+
+    /**
+     * Used to get the plural name of the default currency.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultPlural() {
+        return tl("moneyPlural");
+    }
+
+    /**
+     * Used to get the singular name of the default currency.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultSingular() {
+        return tl("moneySingular");
+    }
+
+    /**
+     * Used to get the plural name of the default currency for a world.
+     * @param world The world to be used in this check.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultPlural(String world) {
+        return tl("moneyPlural");
+    }
+
+    /**
+     * Used to get the singular name of the default currency for a world.
+     * @param world The world to be used in this check.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultSingular(String world) {
+        return tl("moneySingular");
+    }
+
+    /**
+     * Checks to see if a {@link Currency} exists with this name.
+     * @param name The name of the {@link Currency} to search for.
+     * @return True if the currency exists, else false.
+     */
+    @Override
+    public boolean hasCurrency(String name) {
+        return name.equalsIgnoreCase(tl("moneySingular"));
+    }
+
+    /**
+     * Checks to see if a {@link Currency} exists with this name.
+     * @param name The name of the {@link Currency} to search for.
+     * @param world The name of the {@link World} to check for this {@link Currency} in.
+     * @return True if the currency exists, else false.
+     */
+    @Override
+    public boolean hasCurrency(String name, String world) {
+        return name.equalsIgnoreCase(tl("moneySingular"));
+    }
+
+    /**
+     * Checks to see if a {@link Currency} exists with this name.
+     * @param name The name of the {@link Currency} to search for.
+     * @return True if the currency exists, else false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncHasCurrency(String name) {
```

Comment:
I'd rather all the async methods were in a separate abstract class, as they're largely just boilerplate wrappers around the sync methods, while the sync methods contain the actual functionality.

Reply:
Furthermore, if these are functionally identical to the sync methods, why duplicate the method code?